saleor/graphql/account/mutations/account.py

```diff
@@ -65,12 +65,20 @@ def mutate(cls, root, info, **data):
 
     @classmethod
     def clean_input(cls, info, instance, data, input_cls=None):
+        if not settings.ENABLE_ACCOUNT_CONFIRMATION_BY_EMAIL:
+            return super().clean_input(info, instance, data, input_cls=None)
+
         try:
             validate_storefront_url(data["redirect_url"])
         except ValidationError as error:
             raise ValidationError(
                 {"redirect_url": error}, code=AccountErrorCode.INVALID
             )
+        except KeyError:
```

Comment:
I don't see the test for lack of "redirect_url" when it's required.

Reply:
I'm not sure but in all mutation, we check if `key` in `input_data` instead of catching `KeyError` and I prefer a solution with the condition. It's more clear to me.  

@maarcingebala  What do you think about it?

Reply:
`try - except` in general have better performance than `if key in` so I'm just used to that syntax for quite a while, however, since `if key in` is the way we do things here maybe it's better to stay consistent.
@maarcingebala thoughts?

Reply:
@fowczarek will add the missing test, good idea.

Reply:
I would say `if-else` is used more often in our codebase in cases like this. But both solutions are perfectly fine and readable in my opinion and I wouldn't bother with changing that.